internal/cli/run.go

```diff
@@ -98,43 +99,23 @@ func runController() {
 		checkf("failed to create output directory: %v", err)
 	}
 
-	experiment := orchestration.Experiment{
-		Logger:      logging.New("ctrl"),
-		NumReplicas: viper.GetInt("replicas"),
-		NumClients:  viper.GetInt("clients"),
-		Duration:    viper.GetDuration("duration"),
-		Output:      outputDir,
-		ReplicaOpts: &orchestrationpb.ReplicaOpts{
-			UseTLS:            true,
-			BatchSize:         viper.GetUint32("batch-size"),
-			TimeoutMultiplier: float32(viper.GetFloat64("timeout-multiplier")),
-			Consensus:         viper.GetString("consensus"),
-			Crypto:            viper.GetString("crypto"),
-			LeaderRotation:    viper.GetString("leader-rotation"),
-			ConnectTimeout:    durationpb.New(viper.GetDuration("connect-timeout")),
-			InitialTimeout:    durationpb.New(viper.GetDuration("view-timeout")),
-			TimeoutSamples:    viper.GetUint32("duration-samples"),
-			MaxTimeout:        durationpb.New(viper.GetDuration("max-timeout")),
-			SharedSeed:        viper.GetInt64("shared-seed"),
-			Modules:           viper.GetStringSlice("modules"),
-		},
-		ClientOpts: &orchestrationpb.ClientOpts{
-			UseTLS:           true,
-			ConnectTimeout:   durationpb.New(viper.GetDuration("connect-timeout")),
-			PayloadSize:      viper.GetUint32("payload-size"),
-			MaxConcurrent:    viper.GetUint32("max-concurrent"),
-			RateLimit:        viper.GetFloat64("rate-limit"),
-			RateStep:         viper.GetFloat64("rate-step"),
-			RateStepInterval: durationpb.New(viper.GetDuration("rate-step-interval")),
-			Timeout:          durationpb.New(viper.GetDuration("client-timeout")),
-		},
-	}
+	numReplicas := viper.GetInt("replicas")
+	numClients := viper.GetInt("clients")
 
-	experiment.Byzantine, err = parseByzantine()
-	checkf("%v", err)
+	cfgPath := viper.GetString("config")
+
+	var cfg *config.HostConfig
+	if cueCfgPath != "" {
+		cfg, err = config.Load(cfgPath)
+		checkf("config error: %v", err)
+	} else {
+		cfg = config.NewLocal(numReplicas, numClients)
+	}
 
 	worker := viper.GetBool("worker")
-	hosts := viper.GetStringSlice("hosts")
+
+	hosts := cfg.AllHosts()
```

Comment:
This returns an empty slice if the cfg (HostConfig) has localhost. Is that intended?